lib/improver/utilities/statistical_operations.py

```diff
@@ -0,0 +1,427 @@
+               threshold values to the percentiles slice; here we assume
+               inverse_ordering is False, so we use >=. We then populate the
+               value_bounds and percentile_bounds arrays.
+
+               Slice 0 - 0th Percentile::
+
+                   [[1.0 >= 2.0, 1.0 >= 2.0, 1.0 >= 2.0],
+                    [3.0 >= 2.0, 3.0 >= 2.0, 3.0 >= 2.0],
+                    [5.0 >= 2.0, 5.0 >= 2.0, 5.0 >= 2.0]]
+
+                   [[False, False, False],
+                    [True, True, True],
+                    [True, True, True]]
+
+               The value_bounds array has a leading dimensions with 2 indices
+               to be associated with the lower [0] and upper bounds [1] about
+               the threshold being considered. The [0] index is populated with
+               the values in the slice of percentiles_cube at every True index.
+               The [1] index is populated with the values in the next slice of
+               percentiles_cube.::
+
+                   [ [[np.nan, np.nan, np.nan],
+                      [2.0, 2.0, 2.0],
+                      [2.0, 2.0, 2.0]],
+
+                     [[np.nan, np.nan, np.nan],
+                      [4.0, 4.0, 4.0],
+                      [4.0, 4.0, 4.0]] ]
+
+               The percentile_bounds array is also contains a leading dimension
+               associated with lower and upper bounds about the thresholds. The
+               lower bound array is populated at every True index with the
+               current percentile value (0 in this first slice), whilst the
+               upper bound array takes the percentile value from the next
+               slice.::
+
+                   [ [[-1, -1, -1],
+                      [0, 0, 0],
+                      [0, 0, 0]],
+
+                     [[-1, -1, -1],
+                      [50, 50, 50],
+                      [50, 50, 50]] ]
+
+               After the same process is applied to the next slice, the 50th
+               percentile, we end up with value_bounds::
+
+                   [ [[np.nan, np.nan, np.nan],
+                      [2.0, 2.0, 2.0],
+                      [4.0, 4.0, 4.0]],
+
+                     [[np.nan, np.nan, np.nan],
+                      [4.0, 4.0, 4.0],
+                      [4.0, 4.0, 4.0]] ]
+
+               And percentile bounds::
+
+                   [ [[-1, -1, -1],
+                      [0, 0, 0],
+                      [50, 50, 50]],
+
+                     [[-1, -1, -1],
+                      [50, 50, 50],
+                      [50, 50, 50]] ]
+
+               Note that where there is no availble +1 index in the
+               percentiles_cube the upper bound is set to be the same as the
+               lower_bound.
+
+            2. When all slices have been interated over, the interpolants are
+               calculated using the threshold values and the values_bounds.
+
+               (threshold_cube.data - lower_bound)/(upper_bound - lower_bound)
```

Comment:
It might be useful if these lines of maths were code blocks too to make it more readable.

Reply:
:white